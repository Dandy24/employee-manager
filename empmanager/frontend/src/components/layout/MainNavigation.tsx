import {Header} from "antd/es/layout/layout";
import {Menu} from "antd";
import MenuItem from "antd/es/menu/MenuItem";
import { Link } from "react-router-dom";

export function MainNavigation(): JSX.Element {

    return(
        <Header>
            <Menu theme='dark' mode='horizontal'>
                <MenuItem key='company-list'><Link to='/'>Seznam firem</Link></MenuItem>
                <MenuItem key='company-create'><Link to='/new-company'>Přidání nové firmy</Link></MenuItem>
            </Menu>
        </Header>
    )
}
