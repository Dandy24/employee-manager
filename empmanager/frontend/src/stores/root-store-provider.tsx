import React from 'react';
import { RootStore } from './root-store'

let rootStore: RootStore;
const RootStoreContext = React.createContext<RootStore>({} as RootStore);

export const useRootStore = (): RootStore => {
    const context = React.useContext(RootStoreContext);
    if (context === undefined) {
        throw new Error('useStore must be used within StoreProvider');
    }

    return context;
};

interface RootStoreProviderProps {
    rootStore: RootStore;
    children: React.ReactNode;
}

export const RootStoreProvider: React.FC<RootStoreProviderProps> = ({
                                                                        rootStore,
                                                                        children,
                                                                    }: RootStoreProviderProps): React.ReactElement => {
    return <RootStoreContext.Provider value={rootStore}>{children}</RootStoreContext.Provider>;
};

// export function initializeRootStore(host: string, initialData?: RootStoreHydration): RootStore {
//     const api = new Api(host, process.env.NODE_ENV === 'production');
//
//     const _rootStore = rootStore ?? new RootStore(host, api);
//
//     // If your page has Next.js data fetching methods that use a Mobx rootStore, it will
//     // get hydrated here, check `pages/ssg.js` and `pages/ssr.js` for more details
//     if (initialData) {
//         _rootStore.rehydrate(initialData);
//     }
//     // For SSG and SSR always create a new rootStore
//     if (typeof window === 'undefined') {
//         return _rootStore;
//     }
//     // Create the rootStore once in the client
//     if (!rootStore) {
//         rootStore = _rootStore;
//     }
//
//     return _rootStore;
// }
