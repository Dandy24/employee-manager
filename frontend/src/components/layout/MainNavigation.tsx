import React from 'react';
import { Menu } from 'antd';
import { Link } from 'react-router-dom';
import { observer } from 'mobx-react-lite';
import { useRootStore } from '../../stores/root-store-provider';
import { Header } from 'antd/lib/layout/layout';
import MenuItem from 'antd/lib/menu/MenuItem';
import { HomeOutlined } from '@ant-design/icons';

export const MainNavigation: React.FC = observer((): JSX.Element => {
    const rootStore = useRootStore();

    return (
        <Header data-testid="main-header">
            <Menu theme="dark" mode="horizontal" selectedKeys={[rootStore.activePage]}>
                <MenuItem key="dashboard" data-testid="menu-dashboard-item">
                    <Link to="/" data-testid="menu-dashboard-item-link">
                        <HomeOutlined />
                    </Link>
                </MenuItem>
                <MenuItem key="company-list" data-testid="menu-company-list-item">
                    <Link to="/company-list" data-testid="menu-company-list-item-link">
                        Seznam firem
                    </Link>
                </MenuItem>
                <MenuItem key="employee-list" data-testid="menu-employee-list-item">
                    <Link to="/employee-list" data-testid="menu-employee-list-item-link">
                        Seznam zaměstnanců
                    </Link>
                </MenuItem>
            </Menu>
        </Header>
    );
});
