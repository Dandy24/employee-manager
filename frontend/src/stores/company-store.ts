import { RootStore } from './root-store';
import { action, makeObservable, observable, runInAction } from 'mobx';
import { createCompany, deleteCompany, getCompanyList, updateCompany } from '../api/apiCalls';
import { message } from 'antd';
import { CompanyEntity } from '../models/entities/company-entity';
import { CompanyDto } from '../models/dtos/company-dto';
import { SearchableCompanyEntity } from '../models/entities/searchable-company-entity';

import 'moment/locale/cs';
import moment from 'moment';

moment.locale('cs');

export class CompanyStore {
    companies: CompanyEntity[] = [];
    company: CompanyEntity;
    loadingCompanies = false;
    isEditOpen = false;
    rootStore: RootStore;

    constructor(rootStore: RootStore) {
        this.rootStore = rootStore;

        makeObservable(this, {
            company: observable,
            companies: observable,
            loadingCompanies: observable,
            isEditOpen: observable,

            getCompaniesFromStorage: action,

            fetchAllCompanies: action,
            closeModal: action,
            openToAdd: action,
            openToEdit: action,
            saveCompany: action,
            updateCompany: action,
            deleteCompany: action,
        });
    }

    getCompaniesFromStorage(): void {
        this.companies.push(JSON.parse(localStorage.getItem('company')));
    }

    // FIXME proc se tady neposila string na BE a tam se nepouziva filter (WHERE atd.)
    async fetchAllCompanies(filter?: string, selected?: SearchableCompanyEntity): Promise<void> {
        runInAction(() => {
            this.loadingCompanies = true;
            this.companies = [];
        });

        try {
            const companies = await getCompanyList();

            let filteredCompanies;

            if (filter) {
                filteredCompanies = companies.filter(
                    (comp) =>
                        filter.toLowerCase().includes(comp.name.toLowerCase()) ||
                        comp.name.toLowerCase().includes(filter.toLowerCase()) ||
                        filter.toLowerCase().includes(comp.address.toLowerCase()) ||
                        comp.address.toLowerCase().includes(filter.toLowerCase()) ||
                        filter.includes(comp.phone.toString()),
                );
            }

            if (selected?.company) {
                filteredCompanies = companies.filter(
                    (comp) => comp.name === selected.company.name || comp.address === selected.company.address,
                );
            }
            runInAction(() => {
                this.companies = filteredCompanies ? filteredCompanies : companies;
            });
        } catch (e) {
            message.error('Failed to load companies from database');
        } finally {
            runInAction(() => {
                this.loadingCompanies = false;
            });
        }
    }

    closeModal(): void {
        this.isEditOpen = false;
    }

    openToAdd(): void {
        runInAction(() => {
            this.isEditOpen = true;
            this.company = new CompanyEntity();
        });
    }

    openToEdit(company: CompanyEntity): void {
        const comp = this.companies.find((c) => c.id === company.id);
        this.isEditOpen = true;
        if (comp) {
            runInAction(() => {
                this.company = comp;
                if (this.company.profile_picture) {
                    this.company.profile_picture = `${
                        !this.company.profile_picture.includes('http://localhost:8000') ? 'http://localhost:8000' : ''
                    }${this.company.profile_picture.toString()}`;
                }
            });
        } else {
            throw new Error(`Spolecnost s ID ${company.id} nenalezena!`);
        }
    }

    async saveCompany(company: CompanyDto): Promise<void> {
        if (this.company.id) {
            try {
                const updatedCompany = await updateCompany(company, this.company.id);
                message.warning('Údaje o společnosti byly upraveny');
                if (updatedCompany) {
                    this.updateCompany(updatedCompany);
                }
                await this.fetchAllCompanies();
                runInAction(() => {
                    this.closeModal();
                });
            } catch (e) {
                message.error('Údaje o společnosti se nepodařilo upravit.');
                console.log(e);
            }
        } else {
            try {
                await createCompany(company);
                message.success('Spolecnost byla uspesne vytvorena');
                await this.fetchAllCompanies();
                runInAction(() => {
                    this.closeModal();
                });
            } catch (e) {
                message.error('Spolecnost se nepodarilo vytvorit');
                console.log(e);
            }
        }
    }

    async deleteCompany(company: CompanyEntity): Promise<void> {
        await deleteCompany(company.id).catch(() => {
            message.error('Firmu se nepodařilo smazat.');
        });
        runInAction(() => {
            this.loadingCompanies = true;
            message.success('Firma byla úspěšně smazána.');
        });
        await this.fetchAllCompanies();
        runInAction(() => {
            this.loadingCompanies = false;
        });
    }

    updateCompany(updatedCompany: CompanyEntity): void {
        runInAction(() => {
            const updatedCompanyIndex = this.companies.findIndex((company) => company.id === updatedCompany.id);
            if (updatedCompanyIndex !== -1) {
                this.companies[updatedCompanyIndex] = updatedCompany;
            }
        });
    }
}
