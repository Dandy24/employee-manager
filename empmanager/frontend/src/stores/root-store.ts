import {EmployeeStore} from "./employee-store";
import {CompanyStore} from "./company-store";


export class RootStore {
    employeeStore: EmployeeStore;
    companyStore: CompanyStore;

    constructor() {
        this.employeeStore = new EmployeeStore(this);
        this.companyStore = new CompanyStore(this);

        // makeObservable(this, {
        //     contactStore: observable,
        // });
    }

    /*hydrate(): RootStoreHydration {
        return {};
    }

    rehydrate(data: RootStoreHydration): void {
        if (data) {
        }
    }*/
}
