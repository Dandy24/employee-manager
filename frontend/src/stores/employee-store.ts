import { RootStore } from './root-store';
import { action, makeObservable, observable, runInAction } from 'mobx';
import {
    createEmployee,
    deleteEmployee,
    getEmployeeList,
    getEmployeeListForCompany,
    updateEmployee,
} from '../api/apiCalls';
import { message } from 'antd';
import { EmployeeDto } from '../models/dtos/employee-dto';
import { EmployeeEntity } from '../models/entities/employee-entity';
import { SearchableEmployeeEntity } from '../models/entities/searchable-employee-entity';

import 'moment/locale/cs';
import moment from 'moment';

const { REACT_APP_TEST_ENV } = process.env;

moment.locale('cs');

export class EmployeeStore {
    employees: EmployeeEntity[] = [];
    employee: EmployeeEntity;
    loadingEmployees = false;
    isEditOpen = false;
    private rootStore: RootStore;

    constructor(rootStore: RootStore) {
        this.rootStore = rootStore;

        makeObservable(this, {
            employees: observable,
            employee: observable,
            loadingEmployees: observable,
            isEditOpen: observable,

            fetchAllEmployees: action,
            closeModal: action,
            openToAdd: action,
            openToEdit: action,
            saveEmployee: action,
            deleteEmployee: action,
        });
    }

    // FIXME proc se tady neposila string na BE a tam se nepouziva filter (WHERE atd.)
    async fetchAllEmployees(
        companyId: number = null,
        filter?: string,
        selected?: SearchableEmployeeEntity | number,
    ): Promise<void> {
        runInAction(() => {
            this.loadingEmployees = true;
            this.employees = [];
        });

        try {
            const employees = await getEmployeeList();

            /** HOTFIX to allow downloading medias from api while running in testserver enviroment **/
            if (REACT_APP_TEST_ENV === 'true') {
                employees.forEach(
                    (emp) => (emp.profile_picture = emp.profile_picture?.replace('/api/media', '/api/test_media')),
                );
            }
            /** **/

            let filteredEmployees;
            let selectedEmployee;

            if (companyId) {
                await getEmployeeListForCompany(companyId).then((data) =>
                    runInAction(() => {
                        this.employees = data;
                    }),
                );
                return;
            }

            if (filter) {
                filteredEmployees = employees.filter(
                    (emp) =>
                        filter.toLowerCase().includes(emp.first_name.toLowerCase()) ||
                        emp.first_name.toLowerCase().includes(filter.toLowerCase()) ||
                        filter.toLowerCase().includes(emp.last_name.toLowerCase()) ||
                        emp.last_name.toLowerCase().includes(filter.toLowerCase()) ||
                        filter.includes(emp.phone.toString()),
                );
            }

            if (this.rootStore.activePage === 'dashboard') {
                if (typeof selected !== 'number' && selected?.employee) {
                    selectedEmployee = employees.find((emp) => emp.id === selected.employee.id);
                } else {
                    selectedEmployee = employees.find((emp) => emp.id === selected);
                }
            } else {
                if (typeof selected !== 'number' && selected?.employee) {
                    filteredEmployees = employees.filter(
                        (emp) =>
                            emp.first_name === selected.employee.first_name ||
                            emp.last_name === selected.employee.last_name,
                    );
                }
            }

            runInAction(() => {
                this.employees = filteredEmployees ? filteredEmployees : employees;
                this.employee = selectedEmployee ? selectedEmployee : null;
            });
        } catch (e) {
            message.error('Failed to load employees from database');
        } finally {
            runInAction(() => {
                this.loadingEmployees = false;
            });
        }
    }

    closeModal(): void {
        this.isEditOpen = false;
    }

    openToAdd(): void {
        runInAction(() => {
            this.employee = new EmployeeEntity();
            this.isEditOpen = true;
        });
    }

    async openToEdit(employee: EmployeeEntity): Promise<void> {
        const emp = this.employees.find((e) => e.id === employee.id);
        this.isEditOpen = true;
        if (emp) {
            await runInAction(async () => {
                this.employee = emp;
                if (this.employee.attachment) {
                    this.employee.attachment = `${
                        !this.employee.attachment.includes('http://localhost:8000') ? 'http://localhost:8000' : ''
                    }${this.employee.attachment.toString()}`;
                }
                if (this.employee.profile_picture) {
                    this.employee.profile_picture = `${
                        !this.employee.profile_picture.includes('http://localhost:8000') ? 'http://localhost:8000' : ''
                    }${this.employee.profile_picture.toString()}`;
                }
            });
        } else {
            throw new Error(`Zamestnanec s ID ${employee.id} nenalezen!`);
        }
    }

    async saveEmployee(employee: EmployeeDto): Promise<void> {
        /** HOTFIX to allow downloading medias from api while running in testserver enviroment **/
        if (REACT_APP_TEST_ENV === 'true') {
            employee.profile_picture = employee.profile_picture?.replace('/api/media', '/api/test_media');
        }
        /** **/
        if (this.employee.id) {
            try {
                // const updatedEmployee = await updateEmployee(this.employee.id, employee);
                await updateEmployee(this.employee.id, employee);
                message.warning('Udaje o zamestnanci byly upraveny');
                await this.fetchAllEmployees();
                runInAction(() => {
                    this.closeModal();
                });
            } catch (e) {
                console.log(e);
                message.error('Nepodařilo se upravit zaměstnance');
            }
        } else {
            try {
                await createEmployee(employee);
                message.success('Zaměstnanec byl úspěšně přidán');
                await this.fetchAllEmployees();
                runInAction(() => {
                    this.closeModal();
                });
            } catch (e) {
                console.log(e);
                message.error('Nepodařilo se vytvorit zaměstnance');
            }
        }
    }

    async deleteEmployee(id: number): Promise<void> {
        try {
            await deleteEmployee(id);
            runInAction(() => {
                this.loadingEmployees = true;
                message.success('Zamestnanec byl úspěšně smazán.');
            });
            await this.fetchAllEmployees();
        } catch (e) {
            message.error('Zaměstnance se nepodařilo smazat.');
        } finally {
            runInAction(() => {
                this.loadingEmployees = false;
            });
        }
    }
}
