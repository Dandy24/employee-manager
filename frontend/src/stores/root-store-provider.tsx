import React from 'react';
import { RootStore } from './root-store';

const RootStoreContext = React.createContext<RootStore>({} as RootStore);

export const useRootStore = (): RootStore => {
    const context = React.useContext(RootStoreContext);
    if (context === undefined) {
        throw new Error('useStore must be used within StoreProvider');
    }

    return context;
};

interface RootStoreProviderProps {
    rootStore: RootStore;
    children: React.ReactNode;
}

export const RootStoreProvider: React.FC<RootStoreProviderProps> = ({
    rootStore,
    children,
}: RootStoreProviderProps): React.ReactElement => {
    return <RootStoreContext.Provider value={rootStore}>{children}</RootStoreContext.Provider>;
};
