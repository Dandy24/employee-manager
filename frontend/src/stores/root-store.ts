import { EmployeeStore } from './employee-store';
import { CompanyStore } from './company-store';
import { action, makeObservable, observable } from 'mobx';
import { ShiftStore } from './shift-store';
import { SearchStore } from './search-store';
import { CalendarStore } from './calendar-store';
import { DashboardStore } from './dashboard-store';

import 'moment/locale/cs';
import moment from 'moment';

moment.locale('cs');

export class RootStore {
    employeeStore: EmployeeStore;
    companyStore: CompanyStore;
    shiftStore: ShiftStore;
    searchStore: SearchStore;
    calendarStore: CalendarStore;
    dashboardStore: DashboardStore;
    activePage = 'dashboard';

    constructor() {
        this.employeeStore = new EmployeeStore(this);
        this.companyStore = new CompanyStore(this);
        this.shiftStore = new ShiftStore(this);
        this.searchStore = new SearchStore(this);
        this.calendarStore = new CalendarStore(this);
        this.dashboardStore = new DashboardStore(this);

        makeObservable(this, {
            activePage: observable,

            setActivePage: action,
        });
    }

    setActivePage(newPage: string): void {
        this.activePage = newPage;
    }
}
