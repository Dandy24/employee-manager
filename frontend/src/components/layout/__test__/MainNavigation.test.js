import { render, screen } from '@testing-library/react';
import React from 'react';
import { MainNavigation } from '../MainNavigation';
import { BrowserRouter } from 'react-router-dom';
import userEvent from '@testing-library/user-event';
import { RootStore } from '../../../stores/root-store';
import { RootStoreProvider } from '../../../stores/root-store-provider';
import { act } from '@testing-library/react';

test('Menu snapshot matches the previous one', async () => {
    render(
        <BrowserRouter>
            <MainNavigation />
        </BrowserRouter>,
    );

    const mainMenu = await screen.findByTestId('main-header');

    expect(mainMenu).toMatchSnapshot();
});

test('Menu displays on page correctly', async () => {
    render(
        <BrowserRouter>
            <MainNavigation />
        </BrowserRouter>,
    );

    const mainMenuEl = await screen.findByTestId('main-header');

    expect(mainMenuEl).toBeInTheDocument();
});

test('Menu items display correctly in menu', async () => {
    render(
        <RootStoreProvider rootStore={new RootStore()}>
            <BrowserRouter>
                <MainNavigation />
            </BrowserRouter>
        </RootStoreProvider>,
    );

    const CompanyListEl = await screen.findByTestId('menu-company-list-item');

    expect(CompanyListEl.textContent).toBe('Seznam firem');
});

test('Menu items link correctly redirects to right page', async () => {
    // Defining missing window.matchMedia() property from JSDOM, to prevent Jest from crashing
    // https://jestjs.io/docs/manual-mocks#mocking-methods-which-are-not-implemented-in-jsdom

    Object.defineProperty(window, 'matchMedia', {
        writable: true,
        value: jest.fn().mockImplementation((query) => ({
            matches: false,
            media: query,
            onchange: null,
            addListener: jest.fn(), // deprecated
            removeListener: jest.fn(), // deprecated
            addEventListener: jest.fn(),
            removeEventListener: jest.fn(),
            dispatchEvent: jest.fn(),
        })),
    });

    render(
        <BrowserRouter>
            <MainNavigation />
        </BrowserRouter>,
    );

    const employeeLinkEl = await screen.findByTestId('menu-employee-list-item-link');

    await act(async () => {
        userEvent.click(employeeLinkEl);
    });

    /** Vyzaduje vyrenderovani cele aplikace + zpusobuje error se storem **/
    // expect(await screen.findByText('ID zamÄ›stnance')).toBeInTheDocument();
    expect(location.pathname).toEqual('/employee-list');
});
