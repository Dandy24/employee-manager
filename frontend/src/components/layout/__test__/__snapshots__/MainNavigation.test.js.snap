// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu snapshot matches the previous one 1`] = `
<DocumentFragment>
  <header
    class="ant-layout-header"
    data-testid="main-header"
  >
    <ul
      class="ant-menu-overflow ant-menu ant-menu-root ant-menu-horizontal ant-menu-dark"
      data-menu-list="true"
      role="menu"
      tabindex="0"
    >
      <li
        class="ant-menu-overflow-item ant-menu-item ant-menu-item-only-child"
        data-menu-id="rc-menu-uuid-test-dashboard"
        data-testid="menu-dashboard-item"
        role="menuitem"
        style="opacity: 1; order: 0;"
        tabindex="-1"
      >
        <span
          class="ant-menu-title-content"
        >
          <a
            data-testid="menu-dashboard-item-link"
            href="/"
          >
            <span
              aria-label="home"
              class="anticon anticon-home"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="home"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M946.5 505L560.1 118.8l-25.9-25.9a31.5 31.5 0 00-44.4 0L77.5 505a63.9 63.9 0 00-18.8 46c.4 35.2 29.7 63.3 64.9 63.3h42.5V940h691.8V614.3h43.4c17.1 0 33.2-6.7 45.3-18.8a63.6 63.6 0 0018.7-45.3c0-17-6.7-33.1-18.8-45.2zM568 868H456V664h112v204zm217.9-325.7V868H632V640c0-22.1-17.9-40-40-40H432c-22.1 0-40 17.9-40 40v228H238.1V542.3h-96l370-369.7 23.1 23.1L882 542.3h-96.1z"
                />
              </svg>
            </span>
          </a>
        </span>
      </li>
      <li
        class="ant-menu-overflow-item ant-menu-item ant-menu-item-only-child"
        data-testid="menu-company-list-item"
        role="menuitem"
        style="opacity: 1; order: 1;"
        tabindex="-1"
      >
        <span
          class="ant-menu-title-content"
        >
          <a
            data-testid="menu-company-list-item-link"
            href="/company-list"
          >
            Seznam firem
          </a>
        </span>
      </li>
      <li
        class="ant-menu-overflow-item ant-menu-item ant-menu-item-only-child"
        role="menuitem"
        style="opacity: 1; order: 2;"
        tabindex="-1"
      >
        <span
          class="ant-menu-title-content"
        >
          <a
            href="/new-company"
          >
            Přidání nové firmy
          </a>
        </span>
      </li>
      <li
        class="ant-menu-overflow-item ant-menu-item ant-menu-item-only-child"
        data-testid="menu-employee-list-item"
        role="menuitem"
        style="opacity: 1; order: 3;"
        tabindex="-1"
      >
        <span
          class="ant-menu-title-content"
        >
          <a
            data-testid="menu-employee-list-item-link"
            href="/employee-list"
          >
            Seznam zaměstnanců
          </a>
        </span>
      </li>
      <li
        class="ant-menu-overflow-item ant-menu-item ant-menu-item-only-child"
        data-testid="menu-new-employee-item"
        role="menuitem"
        style="opacity: 1; order: 4;"
        tabindex="-1"
      >
        <span
          class="ant-menu-title-content"
        >
          <a
            data-testid="menu-new-employee-item-link"
            href="/new-employee"
          >
            Přidání nového zaměstnance
          </a>
        </span>
      </li>
      <li
        aria-hidden="true"
        class="ant-menu-overflow-item ant-menu-overflow-item-rest ant-menu-submenu ant-menu-submenu-horizontal"
        role="none"
        style="opacity: 0; height: 0px; overflow-y: hidden; order: 9007199254740991; pointer-events: none; position: absolute;"
      >
        <div
          aria-controls="rc-menu-uuid-test-rc-menu-more-popup"
          aria-expanded="false"
          aria-haspopup="true"
          class="ant-menu-submenu-title"
          data-menu-id="rc-menu-uuid-test-rc-menu-more"
          role="menuitem"
          tabindex="-1"
        >
          <span
            aria-label="ellipsis"
            class="anticon anticon-ellipsis"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="ellipsis"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M176 511a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0z"
              />
            </svg>
          </span>
          <i
            class="ant-menu-submenu-arrow"
          />
        </div>
      </li>
    </ul>
    <div
      aria-hidden="true"
      style="display: none;"
    />
  </header>
</DocumentFragment>
`;
